interfaces/system_key.go

```diff
@@ -120,7 +103,7 @@ func generateSystemKey() (*systemKey, error) {
 	sk := &systemKey{
 		Version: 1,
 	}
-	snapdPath, err := cmd.InternalToolPath("snapd")
+	snapdPath, err := findSnapdPath()
```

Comment:
Any reason for the indirection?

Reply:
There are some already existing unit tests targeting findSnapdPath. Maybe I could drop those, provided the test cases are covered in cmd.InternalToolPath tests